apps/emqx_bridge_s3/src/emqx_bridge_s3_connector.erl

```diff
@@ -372,6 +379,13 @@ run_aggregated_upload(InstId, Records, #{aggreg_id := AggregId}) ->
             {error, {unrecoverable_error, Reason}}
     end.
 
+render_records(Records) ->
+    try
+        [unicode:characters_to_binary(R) || R <- Records]
```

Comment:
Q: `Records` are maps 99.9% of time, productions of a Rule SQL expression, to be more specific. I wonder why do we attempt to stringify them like that first?

Reply:
You are right. This was done for non-aggregate mode so I assumed we should do the same for aggregate mode as well but I see now that this is incorrect. In non-aggregate mode we render a template. I will fix. Btw, I accidentally pushed a commit for another issue to this PR. Do you think you can review that one as well so we can merge both with the same PR?

Reply:
Removed the formatting here: [89b47e8](https://github.com/emqx/emqx/pull/13100/commits/89b47e8ffccbeff5b9149f3fc2ab9866d4e611fa)